proposals/inheritdoc.md

```diff
@@ -0,0 +1,201 @@
+﻿# inheritdoc
+
+* [x] Proposed
+* [ ] Prototype: [Complete](https://github.com/sharwell/roslyn/inheritdoc)
+* [ ] Implementation: [In Progress](https://github.com/dotnet/roslyn/BRANCH_NAME)
+* [ ] Specification: [Not Started](pr/1)
+
+## Summary
+[summary]: #summary
+
+<!-- One paragraph explanation of the feature. -->
+
+This feature allows members to inherit documentation from other members, either in whole or in part.
+
+## Motivation
+[motivation]: #motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected outcome? -->
+
+This feature reduces the need to copy/paste documentation comments over time. The primary advantages of the feature relate to overall maintainability of software. With fewer duplicates of documentation, the process for updating documentation in response to a change in the source code is simplified.
+
+## Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the proposal. Explain the design in enough detail for somebody familiar with the language to understand, and for somebody familiar with the compiler to implement,  and include examples of how the feature is used. This section can start out light before the prototyping phase but should get into specifics and corner-cases as the feature is iteratively designed and implemented. -->
+
+I believe a good way to start here would be examining the features currently provided by SHFB for this documentation element, and then identifying ones which would be *impractical* from an implementation perspective or "not particularly needed" from an overall usage perspective.
+
+:link: [inheritdoc](http://ewsoftware.github.io/XMLCommentsGuide/html/86453FFB-B978-4A2A-9EB5-70E118CA8073.htm) (Sandcastle XML Comments Guide)
+
+### Syntax
+
+I believe it's reasonable to use the same form as defined by SHFB:
+
+```xml
+<inheritdoc [cref="member"] [path="xpath-filter-expr"] />
+```
+
+This element may be placed either as a top-level element or as an inline element in a documentation comment.
+
+### Compilation
+
+The compiler is updated in the following ways to account for this element:
+
+* The compiler MUST allow the use of an `inheritdoc` element as a top-level element in XML documentation comments.
+* The compiler MUST allow the use of an `inheritdoc` element as an inline element in XML documentation comments.
+* The compiler SHOULD report a warning if the `inheritdoc` element appears without a `cref` attribute, and no candidate for inheriting documentation exists.
+* The compiler SHOULD emit the documentation file with `inheritdoc` elements replaced by their inherited content. The compiler MAY support emitting the documentation file with `inheritdoc` elements not replaced; in this case the compiler SHOULD preserve the placement and form of the `inheritdoc` element in the XML documentation file, except with the `cref` attribute expanded to the documentation ID of the referenced member.
```

Comment:
>elements replaced by their inherited content [](start = 68, length = 44)

This seems like a bad idea for a number of reasons:
1. The compiler doesn't normally consume doc comments in any way.  This would force the compiler to do so.
2. It makes the compiler output documentation frozen (copy of the referenced doc) at the time the code was compiled, rather than being flexible when typos are corrected in the inherited documentation later.
3. It makes the determinism and build story somewhat more complicated (as the compilation may depend on the doc comments of referenced libraries)

I suggest instead that the compiler should simply copy the inheritdoc element into the output.  Consuming tools would be required to process them.

Reply:
Here are the two main comments covering this:
https://github.com/dotnet/csharplang/issues/313#issuecomment-322226397
https://github.com/dotnet/csharplang/issues/313#issuecomment-322227985

In general, preserving the elements would produce better results when using tools that support the elements, but would produce worse results with tools that have not yet been updated.